src/Components/FollowButton/FollowArtistButton.tsx

```diff
@@ -56,31 +57,42 @@ export class FollowArtistButton extends React.Component<Props> {
   handleFollow = () => {
     const { artist, currentUser, relay, onOpenAuthModal } = this.props
 
+    const newFollowCount = artist.is_followed
+      ? artist.counts.follows - 1
+      : artist.counts.follows + 1
     if (currentUser && currentUser.id) {
       commitMutation(relay.environment, {
         mutation: graphql`
           mutation FollowArtistButtonMutation($input: FollowArtistInput!) {
             followArtist(input: $input) {
               artist {
                 is_followed
+                counts {
+                  follows
+                }
               }
             }
           }
         `,
         variables: {
-          input: {
-            artist_id: artist.id,
-            unfollow: artist.is_followed,
-          },
+          input: { artist_id: artist.id, unfollow: artist.is_followed },
         },
         optimisticResponse: {
           followArtist: {
             artist: {
               __id: artist.__id,
               is_followed: !artist.is_followed,
+              counts: { follows: newFollowCount },
```

Comment:
This is enough to see the number in the header change immediately upon pressing the button!